BACKENDS.md

```diff
@@ -61,6 +61,122 @@ When `mode` is `legacy`, the graphite backend will emit metrics with the followi
+seconds.
+
+### Settings
+The following settings apply to all versions:
+- `api-endpoint`: the API endpoint to use, typically a base address such as `http://influxdb.local`.  The path is
+  automatically determined from the `api-version`, so a path should only be present if there is a reverse proxy setup.
+  Required with No default.
+- `api-version`: the version to use, either `1` or `2`.  Defaults to `2`
+- `compress-payload`: specifies whether or not to compress metrics before sending.  Defaults to `true`.
+- `credentials`: the credentials to use.  This will be provided via an `Authorization: Token <value>` header.  It is
+  not http basic authentication (it is `Token`, not `Basic`), nor does it support JWT with a shared secret.  Please
+  raise an issue if this is desired.  Not required, default is no authentication.
+- `max-request-elapsed-time`: the maximum amount of time to retry before giving up and dropping data, defaults to `15s`
+- `max-requests`: the maximum number of parallel requests.  This is primarily network I/O, with very little CPU, it
+  should be capped if it is overwhelming the influxdb server.  Defaults to 10 times the number of logical cores.
+- `metrics-per-batch`: the number of metrics to send per request.  InfluxDB recommends 5-10k for 1.x and 5k for 2.x.
+  Defaults to `5000`.
+- `transport`: the HTTP transport to use, see [TRANSPORT.md](TRANSPORT.md) for further information.
+
+##### Example configuration
+```toml
+[influxdb]
+api-endpoint='https://influxdb.local/'
+compress-payload=true
+credentials='metrics:metrics'
+max-request-elapsed-time='15s'
+max-requests=20
+metrics-per-batch=7500
+transport='default'
+```
+
+#### Version 1 specific
+- `consistency`: the consistency to send in an enterprise deployment.  Should be `any`, `one`, `quorum`, `all`, or
+  unset.  Any other value will show a warning on startup but still use the provided value.  Defaults to unset.
+- `database`: the database to send to.  Required, no default.
+- `retention-policy`: the retention policy in the database to use.  Not required, default is unset.
+
+##### Example
+```toml
+[influxdb]
+... common settings ...
+
+api-version=1
+consistency='any'
+database='mydatabase'
+retention-policy='myrp'
+```
+
+
+#### Version 2 specific
+- `bucket`: the bucket to send to.  Required, no default.
+- `org`: the org to send to.  Required, no default.
+
+##### Example
+```toml
+[influxdb]
+... common settings ...
+
+api-version=2
+bucket='mydatabase/myrp'
+org=''
+```
+
+Using v1 settings in v2 or v2 settings in v1 will display a warning on startup and ignore the setting.
+
+
+### Tag normalization
+As the dogstatsd protocol doesn't have exact parity with InfluxDBs line protocol, there is some cleanup applied to the
+tags before sending them.  Specifically:
+- tags with only a value are normalized as `unnamed:value`
+- multiple values for duplicate keys are sorted
+- keys with multiple values have their values concatenated with `__`
```

Comment:
Is this something that we would consider making configurable in future?

Reply:
If someone comes up with a reason to have multiple implementations, sure.  What I'm hoping is that someone more familiar with InfluxDB raises an issue to say "oh this has a best practice / standard way that everyone knows, check this link", and then we just move to that and tell anyone else they should follow best practices.

Ideally we'll only ever have one way, to keep things simple.